backend/src/product/product.service.ts

```diff
@@ -35,7 +43,29 @@ export class ProductService {
         console.log(productDto);
     }
 
-    getTrackingList() {}
+    async getTrackingList(userId: string): Promise<TrackingProductDto[]> {
+        const trackingProductList = await this.trackingProductRepository.find({
+            where: { userId: userId },
+            relations: ['product'],
+        });
+        if (trackingProductList.length === 0) {
+            throw new HttpException('상품 목록을 찾을 수 없습니다.', HttpStatus.NOT_FOUND);
+        }
+        const trackingListInfo = trackingProductList.map((product) => {
+            const { productName, productCode, shop, shopUrl, imageUrl } = product.product;
+            return {
+                productName,
+                productCode,
+                shop,
+                shopUrl,
+                imageUrl,
+                targetPrice: product.targetPrice,
+                price: 1234, // 임시 더미 가격 데이터
+            };
+        });
```

Comment:
product.product라는 용어가 조금 어색해보이네요. 구조 분해 할당을 활용해보면 어떨까요?
```suggestion
        const trackingListInfo = trackingProductList.map(({product, targetPrice}) => {
            const { productName, productCode, shop, shopUrl, imageUrl } = product;
            return {
                productName,
                productCode,
                shop,
                shopUrl,
                imageUrl,
                targetPrice,
                price: 1234, // 임시 더미 가격 데이터
            };
        });
```